xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/CompareVersionsTest.java

```diff
@@ -162,11 +162,47 @@ public static void tearDownClass() throws Exception
         getDriver().navigate().refresh();
     }
 
+    @Test
+    public void testCompareVersionScenario()
```

Comment:
It's not a huge fan of huge single test which validate very different stuff.

Reply:
I didn't see another way to ensure the test for deleting version was executing last.
+ apparently it's a recommended practice for functional test:
> Write scenarios (i.e. write only a few test methods, even only one if you can so that you can have a single fixture) instead of a lot of individual tests as you'd do for unit tests.

(source: https://dev.xwiki.org/xwiki/bin/view/Community/Testing#HFunctionalTesting)

Reply:
I'm a big fan for the reasons we describe in our best practices at https://dev.xwiki.org/xwiki/bin/view/Community/Testing#HBestpractices-1

> Since functional tests take a long time to execute (XWiki to start, Browser to start, navigation, etc) it's important to write tests that execute as fast as possible. Here are some tips:
> Write scenarios (i.e. write only a few test methods, even only one if you can so that you can have a single fixture) instead of a lot of individual tests as you'd do for unit tests.

Reply:
It's also much closer to mimick what the user does when using XWiki. They never stop and recreate an XWiki instance. They just do a single scenario. So this allows to be closer to the reality and can allow to find issues that are caused by the sequential execution of stuff.